frame/dynamic-fee/src/lib.rs

```diff
@@ -67,12 +78,13 @@ decl_module! {
 			}
 		}
 
-		#[weight = T::DbWeight::get().writes(1)]
-		fn note_min_gas_price_target(
+		#[weight = (T::DbWeight::get().writes(1), DispatchClass::Mandatory)]
+		pub fn note_min_gas_price_target(
 			origin,
 			target: U256,
 		) {
 			ensure_none(origin)?;
+			ensure!(TargetMinGasPrice::get().is_none(), Error::<T>::NotedMinGasPriceMoreThanOnce);
```

Comment:
Actually, please change this to an `assert!` (and add a comment why the panic here is okay). If a block author pushes a block with multiple noting, then we'd rather want to reject that block instead of throwing an error and continue.

Reply:
@sorpaas If a block author pushes a block with multiple noting, what will happen to the block author's client, will its client panic or just drop the block and reselect another block without multiple  noting?

What if all the block producer's client do the same thing, and the block candidates all have multiple noting, will the chain be stalled then.

More conversations here especially about the cases of runtime upgrade:

https://github.com/darwinia-network/darwinia-common/pull/817#discussion_r710039502

Reply:
It will drop the block. If all block producers do the same thing, the chain will stall.